src/components/plot/user-controls-grid.tsx

```diff
@@ -33,27 +37,123 @@ const UserControlsGrid = ({
+    if (invalidControlData?.length > 0) {
+      alertFunctions.addAlert(`Unable to display the local control with Uid(s): ${invalidControlData.join(', ')} because neither the request_info.data nor the request_info.url_pattern were found`, ChaiseAlertType.WARNING);
+    }
+    setLocalControlData(validatedUserControls);
+
+    if (userControlData?.gridConfig?.layouts && Object.values(userControlData.gridConfig?.layouts)?.length > 0) {
       const mappedLayoutValues = Object.values(userControlData.gridConfig?.layouts)?.map((resLayout: any) => (
         resLayout.map((item: LayoutConfig) => convertKeysSnakeToCamel(({
           // i defines the item on which the given layout will be applied
           i: item?.source_uid,
           ...item,
         })))
       ));
-
-      setLayout(Object.fromEntries(Object.entries(userControlData.gridConfig?.layouts).map(([key]: any, index) => [key, mappedLayoutValues[index]])));
+      let tempLayout;
+      tempLayout = Object.fromEntries(Object.entries(userControlData.gridConfig?.layouts).map(([key]: any, index) => [key, mappedLayoutValues[index]]));
+      const invalidLayoutUid: string[] = [];
+
+      // Check if the layout parameter has all the necessary properties such as source_uid, h, w, x, and y. If any of these properties are missing, display a warning.
+      tempLayout = Object.fromEntries(Object.entries(tempLayout).map(([layoutBPKey, layoutBPValue]) => {
+        const newValue: Layout[] = (layoutBPValue as any[]).filter((val: any) => {
+          if (!val?.sourceUid) {
+            alertFunctions.addAlert(`Unable to display the local control due to a missing 'source_uid' in layouts parameter for "${layoutBPKey}" breakpoint`, ChaiseAlertType.WARNING);
+            return false;
+          }
+          //Not showing single error in this case as it's specific to a breakpoint
+          if (!(val?.w !== 0 && val?.w !== undefined && val?.h !== 0 && val?.h !== undefined && val?.x !== undefined && val?.y !== undefined)) {
+            alertFunctions.addAlert(`Unable to display the local control with Uid "${val.sourceUid}" due to a one of the required layout parameters missing(w, h, x, y) for "${layoutBPKey}" breakpoint`, ChaiseAlertType.WARNING);
+            invalidLayoutUid.push(val.sourceUid);
+            return false;
+          }
+          return true;
+        });
+        return [layoutBPKey, newValue];
+      }));
+
+      //Revalidate to filter out controls with invalid layout
+      const revalidatedUserControls = validatedUserControls?.map((control: UserControlConfig) => {
+        let isControlValid = true;
+        if (invalidLayoutUid.includes(control?.uid)) {
+          isControlValid = false;
+          return ({ ...control, visible: isControlValid });
+        }
+        return ({ ...control });
+      }).filter((control) => control.visible || control?.visible == undefined);
+      setLocalControlData(revalidatedUserControls);
```

Comment:
Don't call this here and instead we will call it after the `else` case below. So we only set the `localControlData` once in this useEffect. See comment below